src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -291,9 +306,35 @@ pub mod tests {
         // Since the key has been deleted, flashback to version 1 should not do
         // anything.
         assert_eq!(
-            must_flashback_write(&engine, k, ts, *ts.incr(), *ts.incr()),
+            must_flashback_to_version(&engine, k, ts, *ts.incr(), *ts.incr()),
             0
         );
         must_get_none(&engine, k, ts);
     }
+
+    #[test]
+    fn test_flashback_to_version_pessimistic() {
+        use kvproto::kvrpcpb::PrewriteRequestPessimisticAction::*;
+
+        let engine = TestEngineBuilder::new().build().unwrap();
+        let k = b"k";
+        let (v1, v2, v3) = (b"v1", b"v2", b"v3");
+        // Prewrite and commit Put(k -> v1) with stat_ts = 10, commit_ts = 15.
+        must_prewrite_put(&engine, k, v1, k, 10);
+        must_commit(&engine, k, 10, 15);
+        // Prewrite and commit Put(k -> v2) with stat_ts = 20, commit_ts = 25.
+        must_prewrite_put(&engine, k, v2, k, 20);
+        must_commit(&engine, k, 20, 25);
+
+        must_acquire_pessimistic_lock(&engine, k, k, 30, 30);
+        must_pessimistic_locked(&engine, k, 30, 30);
+
+        // Flashback to version 17 with start_ts = 30, commit_ts = 40.
+        assert_eq!(must_flashback_to_version(&engine, k, 17, 30, 40), 2);
```

Comment:
In other words, I wonder if the `start_ts = lock_ts = 30` here is guaranteed by the client.

Reply:
Let's use a start_ts other than 30 here

Reply:
Thx for the reminder! It has been revised.